bundles/org.smarthomej.binding.tuya/src/main/java/org/smarthomej/binding/tuya/internal/handler/TuyaDeviceHandler.java

```diff
@@ -0,0 +1,405 @@
+
+    private @Nullable ScheduledFuture<?> reconnectFuture;
+    private boolean disposing = false;
+
+    private final Map<Integer, String> dpToChannelId = new HashMap<>();
+    private final Map<Integer, List<String>> dp2ToChannelId = new HashMap<>();
+    private final Map<String, ChannelTypeUID> channelIdToChannelTypeUID = new HashMap<>();
+    private final Map<String, ChannelConfiguration> channelIdToConfiguration = new HashMap<>();
+
+    public TuyaDeviceHandler(Thing thing, @Nullable List<SchemaDp> schemaDps, Gson gson,
+            SimpleDynamicCommandDescriptionProvider dynamicCommandDescriptionProvider, EventLoopGroup eventLoopGroup,
+            UdpDiscoveryListener udpDiscoveryListener) {
+        super(thing);
+        this.gson = gson;
+        this.udpDiscoveryListener = udpDiscoveryListener;
+        this.eventLoopGroup = eventLoopGroup;
+        this.dynamicCommandDescriptionProvider = dynamicCommandDescriptionProvider;
+        this.schemaDps = Objects.requireNonNullElse(schemaDps, List.of());
+    }
+
+    @Override
+    public void processDeviceStatus(Map<Integer, Object> deviceStatus) {
+        logger.trace("'{}' received status message '{}'", thing.getUID(), deviceStatus);
+
+        deviceStatus.forEach((dp, value) -> {
+            String channelId = dpToChannelId.get(dp);
+            if (channelId != null) {
+
+                ChannelConfiguration configuration = channelIdToConfiguration.get(channelId);
+                ChannelTypeUID channelTypeUID = channelIdToChannelTypeUID.get(channelId);
+
+                if (configuration == null || channelTypeUID == null) {
+                    logger.warn("Could not find configuration or type for channel '{}' in thing '{}'", channelId,
+                            thing.getUID());
+                    return;
+                }
+
+                if (CHANNEL_TYPE_UID_COLOR.equals(channelTypeUID)) {
+                    updateState(channelId, ConversionUtil.hexColorDecode((String) value));
+                } else if (CHANNEL_TYPE_UID_STRING.equals(channelTypeUID)) {
+                    updateState(channelId, new StringType((String) value));
+                } else if (CHANNEL_TYPE_UID_DIMMER.equals(channelTypeUID)) {
+                    updateState(channelId, ConversionUtil.brightnessDecode((double) value, 0, configuration.max));
+                } else if (CHANNEL_TYPE_UID_NUMBER.equals(channelTypeUID)) {
+                    updateState(channelId, new DecimalType((double) value));
+                } else if (CHANNEL_TYPE_UID_SWITCH.equals(channelTypeUID)) {
+                    updateState(channelId, OnOffType.from((boolean) value));
+                }
+            } else {
+                // try additional channelDps, only OnOffType
+                List<String> channelIds = dp2ToChannelId.get(dp);
+                if (channelIds == null) {
+                    logger.warn("Could not find channel for dp '{}' in thing '{}'", dp, thing.getUID());
```

Comment:
```suggestion
                    logger.debug("Could not find channel for dp '{}' in thing '{}'", dp, thing.getUID());
```

Unknown DPTs should only be logged at DEBUG